pkg/apiserver/resthandler.go

```diff
@@ -676,6 +676,13 @@ func UpdateResource(r rest.Updater, scope RequestScope, typer runtime.ObjectType
 			}
 		}
 
+		canDelete := func() bool {
```

Comment:
@smarterclayton If we do something like you suggest, I imagine we'll evolve a bunch of code in an admission controller that has almost exactly the same structure as the code in strategy handlers & validation functions, except it'll all be super generic, operate on untyped objects, and be generally unreadable, adding an Nth layer to our already layer-heavy stack.

Despite that, I imagine it is probably the way to go eventually if we want to let people write arbitrarily complex policies.

But for the moment I'd really like to not solve the general problem. Can you stomach this solution for the time being? It should be easy to transition when we have a true field-level policy engine.